client/src/server.jsx

```diff
@@ -0,0 +1,123 @@
+import { createServer, Model, hasMany, belongsTo } from "miragejs"
+
+export function makeServer({ environment = "test" } = {}) {
+  function genAI(text){
+    return "AI Waiting To answer" + text;
+  }
+
+  let server = createServer({
+    environment,
+
+    models: {
+      user: Model.extend({
+        threads: hasMany()
+      }),
+      thread: Model.extend({
+        user: belongsTo()
+      }),
+    },
+
+    seeds(server) {
+      const bob = server.create("user", { name: "Bob" })
+      const alice = server.create("user", { name: "Alice" })
+
+      server.create("threads", { 
```

Comment:
Your model is called `thread` instead of `threads`, the API endpoint will use plural noun automatically.

```suggestion
      server.create("thread", { 
```